internal/cmd/stack/enable.go

```diff
@@ -0,0 +1,52 @@
+package stack
+
+import (
+	"fmt"
+
+	"github.com/shurcooL/graphql"
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+	"github.com/urfave/cli/v2"
+)
+
+type stackEnableMutation struct {
+	Stack struct {
+		ID       string `graphql:"id"`
+		Disabled bool   `graphql:"isDisabled"`
+	} `graphql:"stackEnable(id: $stack)"`
+}
+type stackDisableMutation struct {
+	Stack struct {
+		ID       string `graphql:"id"`
+		Disabled bool   `graphql:"isDisabled"`
+	} `graphql:"stackDisable(id: $stack)"`
+}
+
+func enable(cliCtx *cli.Context) error {
+	stackID := cliCtx.String(flagStackID.Name)
+
+	if nArgs := cliCtx.NArg(); nArgs != 0 {
+		return fmt.Errorf("expected zero arguments but got %d", nArgs)
+	}
+
+	var mutation stackEnableMutation
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+	}
+
+	return authenticated.Client.Mutate(cliCtx.Context, &mutation, variables)
+}
+
+func disable(cliCtx *cli.Context) error {
+	stackID := cliCtx.String(flagStackID.Name)
+
+	if nArgs := cliCtx.NArg(); nArgs != 0 {
+		return fmt.Errorf("expected zero arguments but got %d", nArgs)
+	}
+
+	var mutation stackDisableMutation
+	variables := map[string]interface{}{
+		"stack": graphql.ID(stackID),
+	}
+
+	return authenticated.Client.Mutate(cliCtx.Context, &mutation, variables)
+}
```

Comment:
How about this? A bit less code duplication.

```go
func enable(cliCtx *cli.Context) error {
	return enableDisable[stackEnableMutation](cliCtx)
}

func disable(cliCtx *cli.Context) error {
	return enableDisable[stackDisableMutation](cliCtx)
}

func enableDisable[T any](cliCtx *cli.Context) error {
	stackID := cliCtx.String(flagStackID.Name)

	if nArgs := cliCtx.NArg(); nArgs != 0 {
		return fmt.Errorf("expected zero arguments but got %d", nArgs)
	}

	var mutation T
	variables := map[string]interface{}{
		"stack": graphql.ID(stackID),
	}

	return authenticated.Client.Mutate(cliCtx.Context, &mutation, variables)
}
```

Reply:
Sounds good - I'm not fussed, easy enough to keep clean. I copied the lock.go code to implement this.